install_test.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright © 2022 - 2024 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_test
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher-sandbox/ele-testhelpers/kubectl"
+	"github.com/rancher-sandbox/ele-testhelpers/rancher"
+	"github.com/rancher-sandbox/ele-testhelpers/tools"
+)
+
+var _ = Describe("Provision k3s cluster and Rancher", Label("install"), func() {
+	// Create kubectl context
+	// Default timeout is too small, so New() cannot be used
+	k := &kubectl.Kubectl{
+		Namespace:    "",
+		PollTimeout:  tools.SetTimeout(300 * time.Second),
+		PollInterval: 500 * time.Millisecond,
+	}
+
+	It("Install upstream k3s cluster", func() {
+		if proxy == "enabled" {
+			By("Run local squid proxy in docker", func() {
+				// Configure proxy before k3s installation if requested
+				GinkgoLogr.Info("Starting squid proxy")
+
+				cwd, _ := os.Getwd()
+				GinkgoLogr.Info("Current working directory: " + cwd)
+
+				out, err := exec.Command("docker", "run", "-d", "--rm", "--name", "squid_proxy",
+					"--volume", cwd+"/.github/scripts/squid.conf:/etc/squid/squid.conf",
+					"-p", "3128:3128", "ubuntu/squid").CombinedOutput()
+				GinkgoWriter.Println(string(out))
+				Expect(err).To(Not(HaveOccurred()))
+			})
+
+			By("Configure proxy in /etc/default/k3s", func() {
+				k3sConfigPath := "/etc/default/k3s"
+				k3sConfig := fmt.Sprintf(`HTTP_PROXY=http://%s
+HTTPS_PROXY=http://%s
+NO_PROXY=127.0.0.0/8,10.0.0.0/8,cattle-system.svc,172.16.0.0/12,192.168.0.0/16,.svc,.cluster.local`, proxyHost, proxyHost)
+				// Write the k3s proxy config file as root
+				out, err := exec.Command("sh", "-c", fmt.Sprintf("echo '%s' | sudo tee %s", k3sConfig, k3sConfigPath)).CombinedOutput()
+				GinkgoWriter.Println(string(out))
+				Expect(err).To(Not(HaveOccurred()))
+			})
+		}
+
+		By("Installing k3s", func() {
+			installCmd := exec.Command("sh", "-c", "curl -sfL https://get.k3s.io | sh -s -")
```

Comment:
What if you initialize and start the server together?
```suggestion
			installCmd := exec.Command("sh", "-c", "curl -sfL https://get.k3s.io | sh -s - server --cluster-init")
```

Reply:
Seems it works, thanks.